core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -0,0 +1,315 @@
+    Namespace namespace = request.namespace();
+    catalog.createNamespace(namespace, request.properties());
+    return CreateNamespaceResponse.builder()
+        .withNamespace(namespace)
+        .setProperties(catalog.loadNamespaceMetadata(namespace))
+        .build();
+  }
+
+  public static GetNamespaceResponse loadNamespace(SupportsNamespaces catalog, Namespace namespace) {
+    Map<String, String> properties = catalog.loadNamespaceMetadata(namespace);
+    return GetNamespaceResponse.builder()
+        .withNamespace(namespace)
+        .setProperties(properties)
+        .build();
+  }
+
+  public static DropNamespaceResponse dropNamespace(SupportsNamespaces catalog, Namespace namespace) {
+    boolean dropped = catalog.dropNamespace(namespace);
+    return DropNamespaceResponse.builder()
+        .dropped(dropped)
+        .build();
+  }
+
+  public static UpdateNamespacePropertiesResponse updateNamespaceProperties(
+      SupportsNamespaces catalog, Namespace namespace, UpdateNamespacePropertiesRequest request) {
+    Set<String> removals = Sets.newHashSet(request.removals());
+    Map<String, String> updates = request.updates();
+
+    Set<String> commonKeys = Sets.intersection(updates.keySet(), removals);
+    if (!commonKeys.isEmpty()) {
+      throw new UnprocessableEntityException(commonKeys);
+    }
+
+    Map<String, String> startProperties = catalog.loadNamespaceMetadata(namespace);
+    Set<String> missing = Sets.difference(removals, startProperties.keySet());
+
+    if (!updates.isEmpty()) {
+      catalog.setProperties(namespace, updates);
+    }
+
+    if (!removals.isEmpty()) {
+      // remove the original set just in case there was an update just after loading properties
+      catalog.removeProperties(namespace, removals);
+    }
+
+    return UpdateNamespacePropertiesResponse.builder()
+            .addMissing(missing)
+            .addUpdated(updates.keySet())
+            .addRemoved(Sets.difference(removals, missing))
+            .build();
+  }
+
+  public static ListTablesResponse listTables(Catalog catalog, Namespace namespace) {
+    List<TableIdentifier> idents = catalog.listTables(namespace);
+    return ListTablesResponse.builder().addAll(idents).build();
+  }
+
+  public static LoadTableResponse stageTableCreate(Catalog catalog, Namespace namespace, CreateTableRequest request) {
+    request.validate();
+
+    TableIdentifier ident = TableIdentifier.of(namespace, request.name());
+    if (catalog.tableExists(ident)) {
+      throw new AlreadyExistsException("Table already exists: %s", ident);
+    }
+
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("created-at", OffsetDateTime.now().toString());
+    properties.putAll(request.properties());
+
+    TableMetadata metadata = TableMetadata.newTableMetadata(
+        request.schema(),
+        request.spec() != null ? request.spec() : PartitionSpec.unpartitioned(),
```

Comment:
I wonder if we should just add these to the "spec() and writeOrder()" methods rather than null checking here.

Reply:
I don't know. I'm reluctant to make the request return defaults instead of `null`. I'd probably leave this as it is here and let the implementation decide how to handle requests without these. That, or update `newTableMetadata` to do it.